spot-contracts/contracts/FeePolicy.sol

```diff
@@ -227,32 +265,78 @@ contract FeePolicy is IFeePolicy, OwnableUpgradeable {
+        ];
+        return (0, vaultFeePerc);
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computePerpToUnderlyingSwapFeePercs() external override returns (uint256, uint256) {
+        // When user swaps perps for vault's underlying -> perps are redeemed by the vault
+        IFeePolicy.SubscriptionState memory s = computeSubscriptionState();
+
+        // Split the fees between the perp and vault systems
+        uint256 vaultFeePerc = vaultSwapFeePerc[
+            keccak256(abi.encodePacked(address(perp), address(vault.underlying())))
+        ];
+        // When the system is over-subscribed, we charge no perp burn fee.
+        if (s.normalizedSubscriptionRatio > 1) {
+            return (0, vaultFeePerc);
+        }
+
+        // When the system is under-subscribed, we charge a perp burn fee on top of the vault's swap fee.
+        return (perpBurnFeePerc, vaultFeePerc);
+    }
+
     /// @inheritdoc IFeePolicy
     function decimals() external pure override returns (uint8) {
         return DECIMALS;
     }
 
-    /// @return If the rollover vault is over-subscribed based on the current tvls.
-    function isOverSubscribed() public returns (bool) {
-        SubscriptionState memory s = computeSubscriptionState();
-        return s.normalizedSubscriptionRatio > ONE;
+    /// @notice Computes the detailed subscription state.
+    /// @return The subscription state.
+    function computeSubscriptionState() public override returns (IFeePolicy.SubscriptionState memory) {
+        return computeSubscriptionState(perp.getDepositBond(), perp.getTVL(), vault.getTVL());
     }
 
-    /// @notice Computes the detailed subscription state based on the current `currentVaultTVL` and `equilibriumVaultTVL`.
+    /// @notice Computes the detailed subscription state based on the recorded `perpTVL` and `vaultTVL`.
     /// @return The subscription state.
-    function computeSubscriptionState() public returns (SubscriptionState memory) {
-        IBondIssuer bondIssuer = perp.bondIssuer();
-        SubscriptionState memory s;
-        s.currentPerpTVL = perp.getTVL();
-        s.currentVaultTVL = vault.getTVL();
-        // NOTE: We assume that perp minting bonds have only 2 tranches and perp assumes the senior one.
+    function computeSubscriptionState(
+        IBondController depositBond,
+        uint256 perpTVL,
+        uint256 vaultTVL
+    ) public view returns (IFeePolicy.SubscriptionState memory) {
+        IFeePolicy.SubscriptionState memory s;
+        s.perpTVL = perpTVL;
+        s.vaultTVL = vaultTVL;
+        // NOTE: We assume that perp only accepts  assumes the senior one.
         //       We assume that perp's TVL and vault's TVL values have the same base denomination.
-        s.equilibriumVaultTVL = s.currentPerpTVL.mulDiv(
-            bondIssuer.trancheRatios(1),
-            bondIssuer.trancheRatios(0),
+        (, uint256 perpTR) = depositBond.tranches(0);
+        uint256 equilibriumVaultTVL = s.perpTVL.mulDiv(
```

Comment:
Say perp TVL is 100 and the bond is [25/75]

eq vault TVL should be 300:
= 100 * (750/250)

so, shouldn't this be:
`s.perpTVL.mulDiv(TRANCHE_RATIO_GRANULARITY - perpTR, perpTR)` ?

Reply:
Good catch, that's right.